crates/miden-objects/src/account/account_id/network_id.rs

```diff
@@ -1,20 +1,60 @@
-use alloc::string::ToString;
+use alloc::string::{String, ToString};
 use core::str::FromStr;
 
 use bech32::Hrp;
 
 use crate::errors::NetworkIdError;
 
+/// A wrapper around HRP for custom network identifiers.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
```

Comment:
I think we could actually derive `Hash` for this as well, and also add it on `NetworkId`.